Celeste.Mod.mm/Mod/Entities/EntityCollider.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Monocle;
+using System;
+using Microsoft.Xna.Framework;
+
+namespace Celeste.Mod.Entities {
+    /// <summary>
+    /// Allows for Collision with any type of entity in the game, similar to a PlayerCollider or PufferCollider.
+    /// Performs the Action provided on collision. 
+    /// </summary>
+    /// <typeparam name="T">The specific type of Entity this component should try to collide with</typeparam>
+    [Tracked(false)]
+    public class EntityCollider<T> : Component where T : Entity {
+        /// <summary>
+        /// Provides a simple way to know the Entity type of the specific Collider without Reflection
+        /// </summary>
+        public readonly Type entityType = typeof(T);
```

Comment:
`public` fields are usually in `PascalCase` instead of `camelCase`.
Also the "without Reflection" sounds a bit weird, because `Type` _is_ reflection.
Maybe just "Type of the entity which is checked against." (with some better wording)?

Reply:
1. This doesn't need to be a field, `=> typeof(T)` would be fine
2. If you have a variable of type `EntityCollider<T>`, then you already need to have known the `T` at compile-time to declare that variable. As far as I can tell, this is only useful when using reflection to access this field, at which point you might as well use reflection to get the generic parameter 🤔

Reply:
My logic is for when getting the component through the tracker instead of instance